app/src/main/java/me/magnum/melonds/common/romprocessors/ZipRomFileProcessor.kt

```diff
@@ -118,7 +118,14 @@ class ZipRomFileProcessor(private val context: Context, private val ndsRomCache:
             } else {
                 val romHash = rom.uri.hashCode().toString()
                 val cachedFile = File(romCacheDir, romHash)
-
+                val iconCacheDir = context.externalCacheDir?.let { File(it, RomIconProvider.ICON_CACHE_DIR) }
```

Comment:
need to have ICON_CACHE_DIR as a public variable or to put it in this file, I haven't done any so you can choose your preferred option.

Reply:
In this case, it's better to move the icon removal logic to the `RomIconProvider`. You can create a method to delete the icon of a specific ROM and not only delete it from the file cache, but also from the memory cache. However, you will need to inject the `RomIconProvider` instance into this class and, subsequently, in the `RomFileProcessorFactoryImpl`. If you are not familiar with Hilt, let me know.

Also, leave a comment in the code explaining that this is a workaround for bad icons.

Reply:
I am not really familiar with Hilt, just tried something but it didn't work...

Reply:
I create a dependency cycle when I try to implement this and I haven't succeeded to solve it with dagger as I completely new to this concept. I will push a commit with what I have (but don't compile)